src/utils/convert.js

```diff
@@ -291,43 +298,75 @@ const writeAuth = (definition, newAppDir) => {
     .then(content => createFile(content, fileName, newAppDir));
 };
 
-// Check if scripting has a given method for a step type, key, and position (pre, post, full)
-const hasScriptingMethod = (js, type, key, position) => {
+// Check if scripting has a given method for a step type, key, position (pre, post, full),
+// and method_type (step, resource, input_fields, output_fields)
+const hasScriptingMethod = (js, type, key, position, method_type = 'step') => {
   if (!js) {
     return false;
   }
 
-  let methodSuffix = '';
-
-  if (type === 'trigger') {
-    if (position === 'pre') {
-      methodSuffix = '_pre_poll';
-    } else if (position === 'post') {
-      methodSuffix = '_post_poll';
-    } else {
-      methodSuffix = '_poll';
-    }
-  } else if (type === 'create') {
-    if (position === 'pre') {
-      methodSuffix = '_pre_write';
-    } else if (position === 'post') {
-      methodSuffix = '_post_write';
-    } else {
-      methodSuffix = '_write';
-    }
-  } else if (type === 'search') {
-    if (position === 'pre') {
-      methodSuffix = '_pre_search';
-    } else if (position === 'post') {
-      methodSuffix = '_post_search';
-    } else {
-      methodSuffix = '_search';
-    }
+  const suffixTable = {
+    trigger: {
+      pre: {
+        step: '_pre_poll',
+        output_fields: '_pre_custom_trigger_fields',
+      },
+      post: {
+        step: '_post_poll',
+        output_fields: '_post_custom_trigger_fields',
+      },
+      full: {
+        step: '_poll',
+        output_fields: '_custom_trigger_fields',
+      },
+    },
+    create: {
+      pre: {
+        step: '_pre_write',
+        input_fields: '_pre_custom_action_fields',
+        output_fields: '_pre_custom_action_result_fields',
+      },
+      post: {
+        step: '_post_write',
+        input_fields: '_post_custom_action_fields',
+        output_fields: '_post_custom_action_result_fields',
+      },
+      full: {
+        step: '_write',
+        input_fields: '_custom_action_fields',
+        output_fields: '_custom_action_result_fields',
+      },
+    },
+    search: {
+      pre: {
+        step: '_pre_search',
+        resource: '_pre_read_resource',
+        input_fields: '_pre_custom_search_fields',
+        output_fields: '_pre_custom_search_result_fields',
+      },
+      post: {
+        step: '_post_search',
+        resource: '_post_read_resource',
+        input_fields: '_post_custom_search_fields',
+        output_fields: '_post_custom_search_result_fields',
+      },
+      full: {
+        step: '_search',
+        resource: '_read_resource',
+        input_fields: '_custom_search_fields',
+        output_fields: '_custom_search_result_fields',
+      },
+    },
+  };
```

Comment:
Refactored the spaghetti code with a mapping table.

Reply:
Much simpler and nice! Good work!